pkg/cli/logout/logout.go

```diff
@@ -110,13 +113,18 @@ func (o LogoutOptions) Validate(args []string) error {
 	}
 
 	if len(o.Config.BearerToken) == 0 {
-		return errors.New("You must have a token in order to logout.")
+		if o.Config.ExecProvider == nil {
+			return errors.New("You must have a token in order to logout.")
+		}
 	}
 
 	return nil
 }
 
 func (o LogoutOptions) RunLogout() error {
+	if o.Config.ExecProvider != nil {
```

Comment:
is it possible to to have a kubeconfig where there are multiple clusters configured with some of them using the execprovider for login and some of them don't?

Reply:
Each context in kubeconfig has it's own login mechanism. Some case use exec plugin and some can use token based.